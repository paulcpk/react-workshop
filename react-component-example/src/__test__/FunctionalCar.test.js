import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Btn from "../FunctionalBtn";

const BUTTON_INITIAL_COUNT = 0;

test("renders without crashing", () => {
  render(<Btn />);
});

test("renders Count: 0", () => {
  // preconditions / setup
  render(<Btn />);
  const btnText = screen.getByText(`Count: 0`);
  // assertion
  expect(btnText).toBeInTheDocument();
});

test("does not render Count: 1", () => {
  // preconditions / setup
  render(<Btn />);
  //expect(() => screen.getByText(`Count: 1`)).toThrow();

  const btnText = screen.queryByText(`Count: 1`);
  // assertion
  expect(btnText).not.toBeInTheDocument();
});

// Snapshot test
it("Btn renders correctly", () => {
  const tree = renderer.create(<Btn />).toJSON();
  expect(tree).toMatchSnapshot();
});
