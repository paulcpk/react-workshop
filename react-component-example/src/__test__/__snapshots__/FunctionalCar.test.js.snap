// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "display": "block",
      "marginBottom": "1rem",
    }
  }
>
  Count: 
  0
</button>
`;
